packages/compass/src/app/setup-plugin-manager.js

```diff
@@ -44,9 +44,7 @@ marky.mark('Loading plugins');
 // eslint-disable-next-line no-nested-ternary
 const COMPASS_PLUGINS = process.env.HADRON_READONLY === 'true'
   ? require('./plugins/readonly')
-  : process.env.HADRON_ISOLATED === 'true'
-    ? require('./plugins/isolated')
```

Comment:
The side-effect of doing this would be that the plugin is mounted on the app start: https://github.com/mongodb-js/compass/blob/b659e8aeb048e2404ce277ca0f86a51d2dcdfefa/packages/compass/src/app/index.js#L171-L177

Should we add a check for the user preferences there now that we are removing a separate plugins list for isolated version?

Reply:
The plugin is mounted, but it never actually reaches the state in which its contents are displayed if `autoUpdates` is `false`, right? We don’t check for updates, so we don’t ever get the `app:update-available` event and consequently `isVisible` remains `false` in the plugin?

Reply:
Yeah, that's true that it doesn't do anything, I was wondering if we want to completely preserve the old behavior where you wouldn't be able to find any references for the autoupdate in the application. But I guess this would also require us to check the `state` to know whether this is just dynamically disabled by the user through options or completely overriden by config and this sounds complex enough to just keep the plugin mounted even if it doesn't do anything

Reply:
Yeah – I’m okay with Isolated/Readonly being implemented that way. Especially Readonly is really just UIs being missing rather than actual backend code being disabled.